brainpy/_src/math/ndarray.py

```diff
@@ -923,6 +922,459 @@ def __dlpack__(self):
+    则返回依次扩展axis[i]的结果，
+    即self.expand_dims(axis)==self.expand_dims(axis[0]).expand_dims(axis[1])...expand_dims(axis[len(axis)-1])
+
+
+    1. If the type of axis is int:
+
+    Returns an Array of dimensions inserted before the axis dimension based on self,
+
+    The first | axis < 0 indicates the bottom axis | dimensions,
+
+    Set n=len(self._value.shape), then axis has the range [-(n+1),n]
+
+
+    2. If the type of axis is Sequence[int] :
+
+    Returns the result of extending axis[i] in sequence,
+
+    self.expand_dims(axis)==self.expand_dims(axis[0]).expand_dims(axis[1])... expand_dims(axis[len(axis)-1])
+
+    """
+    return Array(jnp.expand_dims(self.value, axis))
+
+  def expand(self, *shape: Union[int, Sequence[int]]) -> 'Array':
+    """
+    Expand an array to a new shape.
+
+    Parameters
+    ----------
+    shape : tuple or int
+        The shape of the desired array. A single integer ``i`` is interpreted
+        as ``(i,)``.
+
+    Returns
+    -------
+    expanded : Array
+        A readonly view on the original array with the given shape. It is
+        typically not contiguous. Furthermore, more than one element of a
+        expanded array may refer to a single memory location.
+    """
+    return Array(jnp.broadcast_to(self._value, shape))
+
+  def expand_as(self, array: Union['Array', jax.Array, np.ndarray]) -> 'Array':
+    """
+    Expand an array to a shape of another array.
+
+    Parameters
+    ----------
+    array : Array
+
+    Returns
+    -------
+    expanded : Array
+        A readonly view on the original array with the given shape of array. It is
+        typically not contiguous. Furthermore, more than one element of a
+        expanded array may refer to a single memory location.
+    """
+    if not isinstance(array, Array):
+      array = Array(array)
+    return Array(jnp.broadcast_to(self.value, array.value.shape))
+
+  def squeeze(self, 
+              axis: Optional[Union[int, Sequence]]=None) -> 'Array':
+    return Array(self.squeeze(axis))
+
+  # def item(self, *args) -> Any:
+  #   return self.value.item(*args)
+
+  def pow(self, index: int):
+    return self._value ** index
+
+  def addr(self,
+            vec1: Union['Array', jax.Array, np.ndarray],
+            vec2: Union['Array', jax.Array, np.ndarray],
+            *,
+            beta: float = 1.0,
+            alpha: float = 1.0,
+            out: Optional[Union['Array', jax.Array, np.ndarray]] = None) -> Union[None, NoReturn]:
+    if not isinstance(beta, int) and not isinstance(beta, float):
+      raise Exception('Wrong beta param of addr')
+    if not isinstance(alpha, int) and not isinstance(alpha, float):
+      raise Exception('Wrong alpha param of addr')
+    if not isinstance(vec1, Array):
+      vec1 = Array(vec1)
+    if not isinstance(vec2, Array):
+      vec2 = Array(vec2)
+    if not isinstance(out, Array):
+      out = Array(out)
+    return _return(brainpy.math.outer(vec1, vec2, out=out))
+
+
+  def addr_(self, 
```

Comment:
pytorch functions that are postfixed with "_" are performing p-computation in place.